src/messages/MessageFlag.hpp

```diff
@@ -65,6 +65,12 @@ enum class MessageFlag : std::int64_t {
     ///  - forsen added "blockedterm" as a blocked term
     ///  - Your message is being checked by mods and has not been sent
     ModerationAction = (1LL << 41),
+    /// The message can't be replied to
+    /// Examples:
+    ///  - message was deleted via single channel chat message deletion (IRC: CLEARMSG, EVENTSUB: channel.chat.message_delete)
+    ///  - message was deleted via chat clear user messages (IRC: CLEARCHAT(user), EVENTSUB: channel.chat.clear_user_messages)
+    /// Note: If this message is inside a reply thread, the root must not have the flag either.
+    InvalidReplyTarget = (1LL << 42),
```

Comment:
I would invert this, calling it something like `UserCanReply`. I think the flags should each be off by default and be enabled if a special condition is met. The default here is that you can't reply to a message (if we're thinking of messages as an item in the chat and not a real message from a chatter).

Reply:
I'm not sure I understand it, but by default each message is reply-able, unless it has been deleted or its reply-thread root was deleted (deleted = deleted as single message, or as timeouted user), or 24 hours has passed since it was sent.

To have this inverted would mean that pretty much every message would almost always have the flag turned on, instead of some special few messages.

Reply:
And before the invalidReplyTarget flag gets resolved, on right click when building the menu, we only consider messages that are replyable

https://github.com/Chatterino/chatterino2/blob/07ad07d5258c5fe26ca7ca5022e4a4b69ae952ab/src/messages/layouts/MessageLayout.cpp#L517-L533